### PR TITLE
TC config - splitting test configuration out by service

### PR DESCRIPTION
**Pull request by megan07**
Creates the TC configuration that will allow us to run tests per package/service. We'll want to combine this in some way with a PR that splits out all the tests, or stub out some tests in each package/service directory.

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Review by SarahFrench — APPROVED** [review]
Approving as this PR generates the same code as what currently exists in a feature branch in the TPGB repo (+ similar for my TPG branch).

I've left a few comments, but they're either some small suggestions about comments to add to files, or they are a review comment made for the benefit of people reading this PR in future and not relevant to us now :)

File: mmv1/third_party/terraform/.teamcity/components/generated/build_components.erb [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import jetbrains.buildServer.configs.kotlin.triggers.schedule
+
+// NOTE: in time this could be pulled out into a separate Kotlin package
+
+// The native Go test runner (which TeamCity shells out to) will fail
+// the entire test suite when a single test panics, which isn't ideal.
+//
+// Until that changes, we'll continue to use `teamcity-go-test` to run
+// each test individually
+const val useTeamCityGoTest = false
```
[the remark this answers is not in the document]
Comment (SarahFrench) [reply mmv1/third_party/terraform/.teamcity/components/generated/build_components.erb:16]: 👆 For benefit of people reading this PR in the future

File: mmv1/third_party/terraform/.teamcity/components/packages.kt
[the remark this answers is not in the document]
Comment (SarahFrench) [reply mmv1/third_party/terraform/.teamcity/components/packages.kt:1]: 👆 For benefit of me reading this PR in the future 😂

File: mmv1/third_party/terraform/.teamcity/components/packages.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+var packages = mapOf(
+        "acctest" to "AccTest",
+        "services" to "Services",
+        "tpgdclresource" to "TPG DCL Resource",
+        "tpgiamresource" to "TPG IAM Resource",
+        "tpgresource" to "TPG Resource",
+        "transport" to "Transport",
+        "verify" to "Verify"
+)
```
Comment (zli82016) [inline mmv1/third_party/terraform/.teamcity/components/packages.kt:11]: `provider` and `fwprovider` are new added packages and need to be added to the list here, I think. They can also be added in a separate PR later.

**Review by zli82016 — APPROVED** [review]
LGTM

**Review by zli82016 — APPROVED** [review]
LGTM based on the comments.

**Comment by SarahFrench** [comment]
I just spotted that this PR is marked as having a contributor who hasn't signed the CLA - this is due to Megan leaving HashiCorp between the time this PR was opened and when the PR will be merged 😢 .
My plan is to ignore that warning, because the CLA issue is referencing Megan's old hashicorp work email address, so I don't think it's something that can be addressed? And the CLA was agreed to at the time of the PR being made.

Though @megan07 you're welcome to sign the CLA as an individual in case you're going to make future contributions 😉

**Review by SarahFrench — APPROVED** [review]

**Comment by SarahFrench** [comment]
Plan is to merge this on Monday, without the service-split PR.

**Pull request merged**